[wip] Adds command for seo check

micheleorselli opened the pull request:
I pushed some code for #6 just to see if I'm moving in the right direction. Adding some comments where I'd like to have some feedback

micheleorselli commented on src/App/KickOff.php:
```diff
@@ -60,8 +74,9 @@ protected function getSections() {
      */
-    protected function handleSection(Section $section) {
-        $url = $section->getTargetUrl();
+    protected function handleSection(Section $section, $targetUrl = null) {
```
Passing $targetUrl here is a quick way to have something working, maybe there are better places to put that, like Configuration, what do you think?

frickelbruder replied:
We can keep that as you developed it. I am not 100% satisfied, but since I can't think of a better approach, it's ok :-)

micheleorselli commented on src/Cli/Commands/SeoCheckCommand.php:
```diff
@@ -0,0 +1,95 @@
+
+
+    private $logs = array(
+        array('name' => 'junit-file', 'shortcut' => 'j', 'mode' => InputOption::VALUE_OPTIONAL, 'description' => 'path to a junit compatible log file'),
```
Are those needed? If the command is meant only to have a quick feedback we can remove that I guess

frickelbruder replied:
Yes I think so, too.

frickelbruder commented:
Hi Michel,

thanks for your work so far. This is definitely the way, it should go. I'll try to comment the code inline.

Don't forget to add some Unit-Tests. ;-)

I really appreciate your work.
Markus

frickelbruder commented on src/App/KickOff.php:
```diff
@@ -48,6 +48,20 @@ public function index($configFile) {
         return $this->errorCount;
     }
 
+    public function seocheck($wepbage)
```
Since this method is nearly a duplicate of the index method I'd prefer to keep one entry point for this class. This would conform best to the SOLID principle.
From the SeoCheckCommand the index-method could be called with the configFile beeing injected. The webpage parameter could be added as optional parameter to the index method.

frickelbruder commented on src/Cli/Commands/SeoCheckCommand.php:
```diff
@@ -0,0 +1,95 @@
+
+    }
+
+    private function handleLogOptions(InputInterface $input) {
```
Yepp, a base class would be great.

frickelbruder commented on src/Configuration/TargetUrl.php:
```diff
@@ -32,4 +32,17 @@ public function getHeaders() {
         return $this->headers;
     }
 
+    public static function fromString($url) {
```
You need some additional checks here, to make sure, that the components are really filled. But I like that method :-)

frickelbruder commented on src/config/Seo.yml:
```diff
@@ -0,0 +1,17 @@
+    target:
+        host: ~
+
+Sections:
```
You must have read my mind. Thats exactly, how I imagined the configuration.

frickelbruder commented:
Ok, seems as if I messed up the review. But do you still get my comments?

micheleorselli commented:
ok, I cleaned up things a bit, in particular:
- removed the logging part in SeoCheckCommand. By doing that there is almost no duplicated code and ad base command class is not needed for now imo
- added some checks and unit tests on TargetUrl::fromString

Let me know if there something more I can do, I will be offline for a week or so starting tomorrow (holidays 🎉)

frickelbruder commented:
Ok, great.

I took it and merged it. 

Thanks for your contribution and happy holidays :-)
